src/contracts/Value/ItemGroupInterface.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\Personalization\Value;
+
+interface ItemGroupInterface
```

Comment:
Since it ultimately represents a collection of items (as shown by `getItems` method), you might want to consider making it an implementation of `Traversable` (allowing both `Iterator` and `IteratorAggregate` to be it's implementations - `Iterator` is rarely used, but still...).

WDYT?

Reply:
Maybe I misunderstand but should I make `ItemGroupInterface` implementation of `Traversable` if `ItemListInterface` is already implementation of `IteratorAggregate` (https://github.com/ezsystems/ezrecommendation-client/commit/585a398b9100d9eec79a7e9afb10a9ff4e9af41c#diff-88be615e5c2de5a164d5f804cdb67ba96663f53589ad779b3db65c5c518c3b87R17) ?

Reply:
From what I see they represent two different things.

`ItemGroupListInterface` is a collection of groups.
`ItemGroupInterface` represents a single group.

So what interfaces they consist on should make no difference for you, mostly.

However, after looking at the interface one more time I have to retract my comment - since `ItemGroupInterface` is not a simple collection and could contain other, collection-like properties, it makes sense to not make it an iterable.

This raises a different question though: `ItemGroupListInterface` should extend `Traversable`, not `IteratorAggregate`. That's because iterables can be represented by both `Iterator` and `IteratorAggregate`, which `Traversable` interface is a sign of (it's an interfaces that cannot be implemented directly, it always comes in one of the above).

While `IteratorAggregate` is the most common form (due to simplicity), `Iterator` is perfectly fine and viable. For one, [Generators](https://www.php.net/manual/en/class.generator.php) (objects created by `yield`) are instances of `Iterator`.